src/test/java/net/datafaker/script/ProvidersDocsGenerator.java

```diff
@@ -0,0 +1,197 @@
+        }
+
+        Optional<JavadocComment> javadocComments = commentsCollection.get().getJavadocComments()
+            .stream()
+            .findFirst();
+
+        if (!javadocComments.isPresent()) {
+            fakersWithoutSinceTag.add(filePath);
+            return "";
+        }
+
+        String comment = javadocComments.get().getContent();
+        boolean isComment = comment.contains("@since");
+
+        if (!isComment) {
+            fakersWithoutSinceTag.add(filePath);
+        }
+
+        return comment;
+    }
+    public String generateColumn(String name, char padSymbol, int length) {
+        if (name.length() >= length) return name;
+        return padSymbol + name + Strings.repeat(String.valueOf(padSymbol), length - name.length() - 1);
+    }
+    /**
+     * Writes header and table header to new 'providers.md'
+     *
+     * @see #writeToFile(String)
+     */
+    private void constructHeaderInProvidersFile() {
+        final int nameColLength = 93;
+        final int descColLength = 132;
+        final int sinceColLength = 7;
+        final String header = "# Fake Data Providers\n" + "\n" + "Datafaker comes with the following list of data providers:" + "\n";
+        final String tableHeader = "\n|" + generateColumn("Name", ' ', nameColLength) + "|" + generateColumn("Description", ' ', descColLength) + "|" + generateColumn("Since", ' ', sinceColLength) + "|\n";
+        final String tableHeaderDelimiter = "|" + generateColumn("", '-', nameColLength) + "|" + generateColumn("", '-', descColLength)  + "|" + generateColumn("", '-', sinceColLength) + "|";
+
+        try {
+            writeToFile(header);
+            writeToFile(tableHeader);
+            writeToFile(tableHeaderDelimiter);
+        } catch (IOException e) {
+            throw new RuntimeException("Error on 'providers.md' file creation", e);
+        }
+    }
+
+    private void writeProviderToTable(final String providerName, final String sinceTag) {
+        String inCompleteProviderEntry = "| " + "[" + addSpaceBetweenNameOfProvider(providerName) + "]" +
+            "({{ datafaker.javadoc }}/" + providerName + ".html)";
+        int lengthOfNameSection = inCompleteProviderEntry.length();
+        // Number of chars in 'Name' section, including first '|'.
+        // '|' in front of 'Description' section not included in length
+        final int totalLengthOfNameSection = 94;
+        int lengthOfNameSectionToFill = totalLengthOfNameSection - lengthOfNameSection;
+
+        final String completeNameSection = "\n" + inCompleteProviderEntry + Strings.repeat(" ", lengthOfNameSectionToFill) + "|";
+
+        final String providerDescriptionSectionEntry = Strings.repeat(" ", 132) ;
+
+        // Format of `@since tag` should be '#.#.#'
+        String providerSinceTagSectionEntry = "|" + " " + sinceTag + " " + "|";
+
+        try {
+            writeToFile(completeNameSection + providerDescriptionSectionEntry + providerSinceTagSectionEntry);
```

Comment:
```suggestion
String nameEntry = "\n|" + generateColumn("[" + addSpaceBetweenNameOfProvider(providerName) + "]" +
            "({{ datafaker.javadoc }}/" + providerName + ".html)", ' ', 93);

        final String providerDescriptionSectionEntry = Strings.repeat(" ", 132) ;

        // Format of `@since tag` should be '#.#.#'
        String providerSinceTagSectionEntry = "|" + " " + sinceTag + " " + "|";

        try {
            writeToFile(nameEntry + providerDescriptionSectionEntry + providerSinceTagSectionEntry);
```

usage of `net.datafaker.script.ProvidersDocsGenerator#generateColumn` could simplify the code here
probably `93` could be extracted into constant since it is also used in `net.datafaker.script.ProvidersDocsGenerator#constructHeaderInProvidersFile`

Reply:
Done